ending_location_visitor.py

```diff
@@ -1,32 +1,259 @@
-    node.end_lineno = node.lineno
-    node.end_col_offset = node.col_offset + len(str(node.value))
+    def type(self, node_type):
+        """Set the type of node that we are inspecting. The type of node is
+        compared against in a private method.
+        """
+        self.node_type = node_type
+        return self;
 
+    def reset(self):
+        """Reset between test methods.
+        """
+        # Keep a list rather than a set, because "too many" is wrong.
+        self.props_check = []
+        self.node_type = None
 
-if __name__ == '__main__':
-    # TODO: turn this into a proper test
-    ending_transformer = TransformVisitor()
 
-    with open('examples/ending_locations/const.py') as f:
-        content = f.read()
 
-    ending_transformer.register_transform(astroid.Const, set_const)
 
-    module = astroid.parse(content)
-    ending_transformer.visit(module)
+################################################################################
+class TestEndingLocation(unittest.TestCase):
+    """
+    Prove the `end_lineno`, `end_col_offset` properties are set on ast nodes.
 
-    ending_visitor = EndingVisitor()
+    Method explanations:
+    ending_transformer.visit(module)
+        Walk the given astroid *tree* and transform each encountered node
+        *** Only the nodes which have transforms registered will actually
+        be replaced or changed.
+    
     ending_visitor.visit(module)
+        Show that the transform worked.
+
+    ending_transformer.easy_register_transform(<astroid-node-name>),
+    i.e.
+    ending_transformer.register_transform(astroid.<Type>, <function>)
+        Register `transform(node)` function to be applied on the given
+        astroid's `node_class`. Internally, appends the function to a
+        dictionary in the TransformVisitor class. The keys are the node types,
+        each node must have been assigned/registered a transform function!!
+        The transform function may return a value which is then used to
+        substitute the original node in the tree.
+        params: node_class, transform function
+
+    We store the correct values as: 
+        [lineno, end_lineno, col_offset, end_col_offset]
+    """
+
+    @classmethod
+    def setUpClass(self):
+        """A class method called before tests in an individual class run.
+        setUpClass is called with the class as the only argument and must be
+        decorated as a classmethod():"""
+        
+        # A visitor to transform the nodes.
+        self.ending_transformer = TransformVisitor()
+
+        # A visitor to test the nodes property correctness.
+        self.ending_visitor = EndingVisitor()
+
+        # Register all `transform(node)` functions here...
+        # TODO: attach more nodes with their transform functions here.
+        self.ending_transformer.register_transform(astroid.Arguments, set_general)
+
+        self.ending_transformer.register_transform(astroid.Const, set_general)
```

Comment:
Please create a separate `register_ending_setters` (or some other name) function in the `setendings.py`, and call it here on the `self.ending_transformer`.

In essence, the creation of the "ending transformer" should be wholly managed from the `setendings.py` module. The latter is in fact is a standalone module, not a helper for this one -- rather, this module is a test module for `setendings.py`.

In a related point, please move this module to inside the `test` folder.